cdap-watchdog/src/main/java/co/cask/cdap/logging/framework/LogFileOutputStream.java

```diff
@@ -85,14 +87,22 @@ long getCreateTime() {
     return createTime;
   }
 
+  /**
+   * get the number of bytes written to output stream
+   * @return file size
+   */
+  long getSize() {
+    return fileSize;
+  }
+
   @Override
   public void flush() throws IOException {
     dataFileWriter.flush();
   }
 
   @Override
   public void sync() throws IOException {
-    flush();
+    fileSize = dataFileWriter.sync();
```

Comment:
oh, wait. Correction. I think we need to call `flush` on `DataFileWriter`, otherwise data may be buffered in the internal binary encoder.